### PR TITLE
Load prefs.json from profile-dir if --profile-dir is specified at launch

### PR DESCRIPTION
**Pull request by mtbentley**
In response to #10098 
Tries to load `prefs.json` from the profile-dir and merge them into the preferences if `--profile-dir` is specified at launch.  The profile-dir preferences take precedence over the default preferences, but command line preferences still take precedence over everything.

Also adds some tests for `prefs.rs`.  These rely on the contents of `resources/prefs.json` (at least `test_get_set_reset_extend()` does), so they may need to be re-worked a bit.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10114)

### CODE REVIEW

**Comment by paulrouget** [comment]
follow up: https://github.com/servo/servo/issues/10338

File: components/util/prefs.rs [hunk 1]
```diff
@@ -145,6 +143,36 @@ fn read_prefs() -> Result<HashMap<String, Pref>, ()> {
...
+pub fn add_user_prefs() {
+    if let Some(ref dir) = opts::get().profile_dir {
+        let mut path = PathBuf::from(dir);
+        path.push("prefs.json");
+        if let Ok(file) = File::open(path) {
+            if let Ok(prefs) = read_prefs_from_file(file) {
+                extend_prefs(prefs);
+            }
+        } else {
+            println!("Error opening prefs.json from profile_dir");
```
Comment (Manishearth) [inline components/util/prefs.rs]: this should be to stderr.

**Comment by Manishearth** [comment]
Looks good to me overall, minor nit

**Comment by matthewbentley** [comment]
I updated it to write to stderr.  I'm not sure if I did it right; I just googled and used the first example.

**Comment by Manishearth** [comment]
Looks good. Do you want to squash any commits?

**Comment by matthewbentley** [comment]
Should I? I could probably squash the first 2 together and second two together.

**Comment by Manishearth** [comment]
That sounds good.

**Comment by matthewbentley** [comment]
Squash'd

**Comment by Manishearth** [comment]
@bors-servo r+

thanks!

**Pull request merged**